tests/Sort.test.ts

```diff
@@ -222,45 +287,179 @@ describe('Sort', () => {
+        expect(
+            Sort.by(
+                {
+                    sorting: [
+                        {
+                            property: 'tag',
+                            reverse: false,
+                            propertyInstance: 0,
+                        },
+                    ],
+                },
+                [t1, t2, t3, t4, t5, t6, t7, t8, t9, t10],
+            ),
+        ).toEqual(expectedOrder);
+    });
+    it('should sort correctly by second tag with no global filter', () => {
+        const t1 = fromLine({ line: '- [ ] a #aaa #jjj' });
+        const t2 = fromLine({ line: '- [ ] a #ggg #ccc' });
+        const t3 = fromLine({ line: '- [ ] a #bbb #iii' });
+        const t4 = fromLine({ line: '- [ ] a #fff #aaa' });
+        const t5 = fromLine({ line: '- [ ] a #ccc #bbb' });
+        const t6 = fromLine({ line: '- [ ] a #eee #fff' });
+        const t7 = fromLine({ line: '- [ ] a #ddd #ggg' });
+        const t8 = fromLine({ line: '- [ ] a #hhh #eee' });
+        const t9 = fromLine({ line: '- [ ] a #iii #ddd' });
+        const t10 = fromLine({ line: '- [ ] a #jjj #hhh' });
+        const expectedOrder = [t4, t5, t2, t9, t8, t6, t7, t10, t3, t1];
         expect(
             Sort.by(
                 {
                     sorting: [
-                        { property: 'tag', reverse: false },
-                        { property: 'done', reverse: false },
+                        {
+                            property: 'tag',
+                            reverse: false,
+                            propertyInstance: 1,
+                        },
                     ],
                 },
-                [one, two, three, four, five],
+                [t1, t2, t3, t4, t5, t6, t7, t8, t9, t10],
             ),
         ).toEqual(expectedOrder);
     });
+    it('should sort correctly by tag defaulting to first with global filter', () => {
+        // Arrange
+        const originalSettings = getSettings();
+        updateSettings({ globalFilter: '#task' });
+
+        const t1 = fromLine({ line: '- [ ] #task a #aaa #jjj' });
+        const t2 = fromLine({ line: '- [ ] #task a #ggg #ccc' });
+        const t3 = fromLine({ line: '- [ ] #task a #bbb #iii' });
+        const t4 = fromLine({ line: '- [ ] #task a #fff #aaa' });
+        const t5 = fromLine({ line: '- [ ] #task a #ccc #bbb' });
+        const t6 = fromLine({ line: '- [ ] #task a #eee #fff' });
+        const t7 = fromLine({ line: '- [ ] #task a #ddd #ggg' });
+        const t8 = fromLine({ line: '- [ ] #task a #hhh #eee' });
+        const t9 = fromLine({ line: '- [ ] #task a #iii #ddd' });
+        const t10 = fromLine({ line: '- [ ] #task a #jjj #hhh' });
+        const t11 = fromLine({ line: '- [ ] #task a' });
+        const t12 = fromLine({ line: '- [ ] #task a #aaaa #aaaa' });
+        const t13 = fromLine({ line: '- [ ] #task a #bbbb ' });
+        const expectedOrder = [
```

Comment:
How were these examples chosen? What value do each of the 13 lines provide in verifying the behaviour of the functionality being tested?

Reply:
@sytone it would be nice to explain the reasoning behind the test cases a bit. I don't think this should prevent us from merging, though @claremacrae.

Reply:
Just made a simple list so it was easy to see sort and then also have multiple combinations of the tags in the sort including tasks with less than the instance chosen and none.

Reply:
Added comments.